### DISCUSSION TITLE
Vacant dev here

### DISCUSSION THREAD

**Discussion by tozpeak** [Aodaruma/coa_tools2#19] [General]
Here I am, the dev who wants to help in resurrection of the COA Tools.

**My motivation**
In a short term, I need a tool to create some mesh animations and to export them for my game. It will most likely use Flutter, and I consider to start making an MIT-licensed runtime for Skeleton animations, which uses an open animation format to work with. Since COA Tools might be the most alive FOSS project in the field, I would like to base my work on it's exported format.

**My goal here**
I would like to support and improve a FOSS 2D skeleton animation tool. The current market state is pretty sad and for an indie game dev who don't need a lot of work to be done, but a simple and free tool to complete their project. There's a Godot Skeleton 2D, but it is made by Godot team for Godot ecosystem, and I feel like generic tool is needed in FOSS field as well.

**My background**
I am NOT an animator. Yet. And I don't have resources to learn and use proprietary tools for my project. I consider to learn COA tools, if they can be used. 
I've worked in a game studio, which heavily used Spine2D. I've made some tooling for animators using existing runtime, so I've consulted with them about their needs.
I worked with Unity for 7 years in several outsource studios, so I have a generic experience of making things work. My specialties are integrating third-party libraries, updating for changed API's and fixing other-people code.
I don't mind to learn programming languages, I've worked with Java, Objective-C, TypeScript whenever I needed it and now I'm learning Dart for my new project.

**Why am I writing this?**
I want you to know, that there is at least one person willing to support you in this project.
I have some free time I would like to spend on this.
I could just fork your repo and try hard to figure all things by myself, maybe even make some fixes, but most probably give up into skipping some animations in my feature-list or going proprietary. You could loose your interest in working solo before I became ready to make my first PR.
But I believe in collaboration. Lets build a community of people who do care about COA Tools to live.
Please, guide me into how I can help and what I should learn first to be helpful ASAP. Until this reaches you, I'll try to guide myself.

**Comment by Aodaruma** [comment]
Thank you for reaching out, tozpeak.

While I am skilled in code analysis and development, I'm not particularly good at gathering and communicating people, focusing on the community, and actively driving its engagement.
However, I truly appreciate the offer to collaborate on development. I'm aware that currently, the coa_tools code isn't widely understood (it took me a long time to analyze it as well). Therefore, I plan to focus on creating a wiki, aiming to make what the codes means in coa_tools more widely known.

Regarding building a community, while I can set up a Discord server, we can communicate publicly using GitHub discussions and issues, so I'm unsure whether to create a server or not.

Currently, I'm prioritizing resolving these issues:
#22
#18
#17
#11
#6
Additionally, I'm currently working on development and fixes for #6. Both #22 and #17 seem like they can be implemented or fixed relatively easily. Would you be able to take on one of them?
If you have any questions about the code or its specifications, please feel free to ask, and I'll be happy to answer.

**Comment by tozpeak** [comment]
Hi! Glad to see your response.

I wasn't asking to drive a community from you personally, I was more like offering my skills. I may help you to keep activity in the repo, so the project felt more alive.
When I found your repo first time, I was a bit skeptical and your sole conversation looked well organized, but frustrating too. It was hard to tell whether the repo is active or abandoned. Then I found a mention of your repo in yet another "is COA Tools dead?" topic somewhere, so decided to at least give it a try. I don't think that I've managed to "lift" a project of this scale on my own, but helping someone to do this is a realistic task for me.

Creating a wiki or code-doc is a great way to make the project more open for further collaborations. I may help you, at least as the curious reader. I'm not familiar much with Blender though, so it could be nice to have some links to the Blender API as well. Also will be useful for new devs, so people with no experience in Blender-plugin-creation were potential contributors.

I'm not sure if Discord server is needed at this scale of team. Documented communication in Issues and Conversations actually will be more helpful for compensating the lack of proper documentation for a while. It could be treated as internal StackOverflow instance :laughing: .
I believe, that Discord server is good for teams at least 10+ people actively working on improving an existing solution. Maybe some day.

I also think, that at some point we can benefit from spreading the word about COA Tools revival. 
Some channels like https://gamefromscratch.com/ would love to write a news about it. At least this channel had very kind opinion about original project and loves to gather attention at FOSS projects. But announcement this big should be done at more stable point, at least with working MVP: "image editor -> COA Tools -> somehow useful exported data".
But we at least can mention that the work is in progress in some random places, where people ask about COA Tools state. Maybe we can gather some more enthusiastic devs.

**Reply by Aodaruma** [reply]
Thank you for replying, and I apologize for the late reply.

Indeed, I had confirmed that PRs were being sent to the original repository, but I didn't get someone continuing it as a fork. I also purchased and tried Live2D, Spine, Moca, etc., but I preferred Blender for its ease of rigging with IK and drivers and ease of animation.
Initially, I intended to launch it entirely as a new personal project for myself, and I thought of it as development only for me. However, I'm grateful that someone is willing to contribute, and although I'm busy, I want to make an effort for this project.

Thanks for the positive response regarding the wiki. it's for taking some time since I don't have much experience writing wiki. If there's a specific part you want to know about, I'd like to write an explanation for that first.

About Discord server, I agree about there aren't many developers at the moment and to hold off for a while.

Thank you for telling me about the forum. Recently, there's been increasing attention to Godot, so it might be a good opportunity to gather developers. However, I don't have much experience or knowledge about collaborative development in OSS, so even if we gather people I don't know what to do in detail.

**Comment by tozpeak** [comment]
I'll have some time in following days to dive into the tasks, so there would be questions about the code. If you have some raw documentation, you may post it as well.
Briefly here:
- #17: My main working machine is on Linux, and I don't have a PS to debug and check my work. However, I've implemented an exporter for GIMP 2.99, which is beta. I don't know about GIMP on Windows, but on Linux stable GIMP relies on Python 2 and some outdated package called gimp-python, which is not supported on my repo. I believe, that original exporter script should work prior to GIMP 2.99, since they introduced Python 3 and major API changes.
- #22: That's what I can try to fix. I wonder, why this don't work, since there are the prefs field in the class for that, but all I can see is my recently added pref. But hey, that's what debugging is about. 😊
- #6: I have some alternative opinion, you might like to hear, more on that in the ticket.

**Reply by Aodaruma** [reply]
Thank you for letting me know specifically what you can do. I'm glad you could help with #22. I also understand that you use GIMP instead of Photoshop. In that case, it might be better to ask you to develop or improve the GIMP script exporter.
I looked at the details of your opinion on #6, thank you. It's not an essential feature, but it's one of the features I personally want in coa_tools2, so I'd like to implement it if I have the time.
